test/class_method_test.ts

```diff
@@ -78,18 +89,88 @@ describe('Merging class methods', () => {
+        .filter(value => value != '');
+      expect(
+        result.filter(res =>
+          /private\s+b\s*\(\s*a\s*:\s*any\s*\)\s*:\s*void\s*\{?/.test(
+            res.toString()
+          )
+        )
+      ).length.to.be.greaterThan(0, 'b should have modifier from base');
     });
     it('the patch if method is present in base and patch, and patchOverride is true.', () => {
       const result: String[] = merge(base, patch, true)
-        .split("\n")
+        .split('\n')
         .map(value => value.trim())
-        .filter(value => value != "");
-      expect(result.filter(res => /public\s+b\s*\(\s*a\s*:\s*any\s*\)\s*:\s*void\s*\{?/.test(res.toString()))).length.to.be.greaterThan(0, 'b should have modifier from patch but was ' + result.reduce((prev, curr) => prev.toString() + curr.toString(), ""));
+        .filter(value => value != '');
+      expect(
+        result.filter(res =>
+          /public\s+b\s*\(\s*a\s*:\s*any\s*\)\s*:\s*void\s*\{?/.test(
+            res.toString()
+          )
+        )
+      ).length.to.be.greaterThan(
+        0,
+        'b should have modifier from patch but was ' +
+          result.reduce((prev, curr) => prev.toString() + curr.toString(), '')
+      );
     });
   });
 
-});
\ No newline at end of file
+  describe('should use async await modifiers', () => {
+    const base = `class a {
+      public async f() {
+        const dialog = this.createDialog();
+        await dialog.present();
+      }
+    }`,
+      patch = `class a {
+        public f() {
+          const dialog = this.createDialog();
+          dialog.present();
+        }
+      }`;
+
+    it('the base if method is present in base and patch.', () => {
+      const result: String[] = merge(base, patch, false)
+        .split('\n')
+        .map(value => value.trim())
+        .filter(value => value != '');
+      expect(
+        result.filter(res =>
+          /public\s+async\s+f\s*\(\s*\)\s*\{?/.test(res.toString())
+        )
+      ).length.to.be.greaterThan(0, 'f should have modifier from base');
+      expect(
+        result.filter(res =>
+          /await\s+dialog\.present\(\)?/.test(res.toString())
+        )
+      ).length.to.be.greaterThan(
+        0,
+        'dialog.present() should have modifier from base'
+      );
+    });
+    it('the patch if method is present in base and patch, and patchOverride is true.', () => {
+      const result: String[] = merge(base, patch, true)
+        .split('\n')
+        .map(value => value.trim())
+        .filter(value => value != '');
+      expect(
+        result.filter(res => /public\s+f\s*\(\s*\)\s*\{?/.test(res.toString()))
+      ).length.to.be.greaterThan(
+        0,
+        'f should have modifier from patch but was ' +
+          result.reduce((prev, curr) => prev.toString() + curr.toString(), '')
+      );
+      expect(
+        result.filter(res => /dialog\.present\(\)?/.test(res.toString()))
```

Comment:
Please check whether this regular expression will return true when having: `dialog.present()` and will return false when having: `await dialog.present()`

Reply:
done